pkg/certificate/manager.go

```diff
@@ -117,9 +118,16 @@ func (m *Manager) start(ctx context.Context, mrcClient MRCClient) error {
 func (m *Manager) handleMRCEvent(mrcClient MRCClient, event MRCEvent) error {
 	switch event.Type {
 	case MRCEventAdded:
-		mrc := event.MRC
+		mrc := event.NewMRC
+
+		// ignore add event if MRC has no state
+		if mrc.Status.State == "" {
+			log.Debug().Msgf("received MRC add event for MRC %s. Ignoring because MRC has no state", mrc.GetName())
```

Comment:
the logging framework allows for keyvalue pairs with Structured logging.  Do we have any guidance on what we publish?  It might be helpful here to have something like mrc=mrcname when debugging down the line

Reply:
That's a great idea! I'm not sure if we have explicit guidelines. We do this for osm error codes already (for example https://github.com/openservicemesh/osm/blob/2fb95a2159bfc75ebc75891d8170ecdad7d74d66/pkg/envoy/lds/ingress.go#L25)

Reply:
cool, I did it in #4952 but not sure if there was any consistency and also saw it in a few other areas of the code base. It would help with operating/debugging if we did define some general guidelines.  for example kubernetes has this: https://github.com/kubernetes/community/blob/a7cb5e7b61ad7e85b60d6beb2170c1edd50a4cb3/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments